### PR TITLE
mergeWithDefaults early return if properties is missing, fixes #205

### LINKED ISSUE #205: Breaking changes in jsonSchema (v0.6.0)
I've noticed that in the v0.6.0 next code doesn't work anymore when I try to insert new record:
```
  static get jsonSchema() {
    return {
      type: 'object',
      required: ['text', 'session_id', 'created_at'],
      // properties: ...
    }
 }
```

I've got next exception:
```
TypeError: Cannot convert undefined or null to object
    at mergeWithDefaults (/vagrant/node_modules/objection/lib/model/ModelBase.js:643:38)
    at UserLogin.$setJson (/vagrant/node_modules/objection/lib/model/ModelBase.js:255:14)
    at UserLogin.$setJson (/vagrant/node_modules/objection/lib/model/Model.js:331:35)
    at Function.fromJson (/vagrant/node_modules/objection/lib/model/ModelBase.js:521:11)
    at Function.ensureModel (/vagrant/node_modules/objection/lib/model/Model.js:607:25)
    at Function.ensureModelArray (/vagrant/node_modules/objection/lib/model/Model.js:632:20)
    at InsertOperation.call (/vagrant/node_modules/objection/lib/queryBuilder/operations/InsertOperation.js:55:40)
    at QueryBuilder.callQueryBuilderOperation (/vagrant/node_modules/objection/lib/queryBuilder/QueryBuilderBase.js:349:19)
    at QueryBuilder.insert (/vagrant/node_modules/objection/lib/queryBuilder/QueryBuilder.js:1098:17)
```

If I add **properties** description it will work correctly.

### PR DESCRIPTION
**Pull request by fl0w**

### CODE REVIEW

File: tests/unit/model/ModelBase.js [hunk 1]
```diff
@@ -374,6 +374,30 @@ describe('ModelBase', function () {
...
+      Model.jsonSchema = {
+        required: ['a']
+      }
+
+      var model = Model.fromJson({a: 100});
+
+      expect(model.a).to.equal(100);
+    });
+
+    it('should not validate if jsonSchema.required exists and jsonSchema.properties == undefined', function () {
```
Comment (koskimas) [inline tests/unit/model/ModelBase.js]: 'should not validate' --> 'should validate' right?

**Comment by koskimas** [comment]
One typo in a test name but otherwise excellent!

File: tests/unit/model/ModelBase.js (hunk 1 above)
Comment (fl0w) [inline tests/unit/model/ModelBase.js]: Right, but it shouldn't pass, expecting a `ValidationError` to be thrown.

**Comment by fl0w** [comment]
Updated the message, did I overreach? :)
Added missing semi-colons (I'm not used to them, maybe add linting/.eslint-rules in the future?).

**Comment by koskimas** [comment]
No, it's fine :) We could add linting at some point when we get more contributors. I don't see a need for it yet. Feel free to add one if it helps you out.

**Comment by fl0w** [comment]
I'm ok by following convention, just don't want to be troubling your package.

**Pull request merged**